crossfire/__init__.py

```diff
@@ -52,15 +58,19 @@ def occurrences(
 
 
 def flatten(data, nested_columns=None):
+    origin_type = type(data)
```

Comment:
Getting this info to use latter, mainly for  `DataFrame` as, for now, I convert to `dict` to flatten and later I will need to return in the same class as it was in the input...

Reply:
Why gatting it now if you might not _use_ it later (e.g. wrong value in `nested_column`)?
Also, why use `isinstance` in some places and compare a type to `origin_type` in other places?

Reply:
> Why gatting it now if you might not _use_ it later (e.g. wrong value in `nested_column`)?

Make sense. I will put after  the `nested_columns` validation, but before the class validation.

> Also, why use isinstance in some places and compare a type to origin_type in other places?

I can change to use only the `origin_type` in those kind of validation. I woun't be able to use `isinstance`, because I might convert pandas to dict.... (I will pretend I didn't read yet the next mensage. ;)